infrastructure/gitauth.py

```diff
@@ -0,0 +1,60 @@
+import os
+import getpass
+import subprocess
+import urllib
+
+# alabno/infrastructure/
+exec_dir = os.path.abspath(os.path.dirname(os.path.abspath(__file__)))
+# alabno/
+alabno_dir = os.path.abspath(exec_dir + os.sep + '..')
+
+conf_file_name = 'gitauth.txt'
+conf_file_path = alabno_dir + os.sep + 'config' + os.sep + conf_file_name
+
+def ask_authentication(force=False):
+    # If forcing, remove existing configuration first
+    if force:
+        cmd = 'rm {}'.format(conf_file_path)
+        subprocess.call(cmd, shell=True)
+    
+    # Check if conf file exists. If yes, exit
+    if os.path.isfile(conf_file_path):
+        return
+    
+    username = ''
+    password = ''
+    print('Please insert your gitlab authentication details. These will be used to clone the student repositories and model answers. If they are left blank, only publicly accessible repositories can be cloned.')
+    print('Username:')
+    username = raw_input()
+    # Need to use URL encoding
+    username = urllib.quote(username)
+    password = getpass.getpass()
```

Comment:
this asks the user to input a password when RunServer is run (similarly to how we ask for the db password)